cdp_scrapers/scraper_utils.py

```diff
@@ -269,3 +269,128 @@ def resolve_person_alias(self, person: Person) -> Person:
         instances.seattle.person_aliases
         """
         return person
+
+    def sanitize_roles(
+        self,
+        roles: Optional[List[Role]],
+        chair_aliases: Optional[List[str]] = None,
+        council_member_aliases: Optional[List[str]] = None,
+        council_president_aliases: Optional[List[str]] = None,
+    ) -> Optional[List[Role]]:
+        """
+        1. Standardize Role.title to preset strings
+        2. Ensure only 1 councilmember Role per term
+
+        Parameters
+        ----------
+        roles: Optional[List[Role]]
+            Person.seat.roles
+
+        chair_aliases: Optional[List[str]]
+            Role.title becomes "Chair" if includes any string in this list
+
+        council_member_aliases: Optional[List[str]]
+            Role.title becomes "Councilmember" if includes any string in this list
+
+        council_president_aliases: Optional[List[str]]
+            Role.title becomes "Council President" if includes any string in this list
+
+        Returns
+        -------
+        roles: Optional[List[Role]]
+            Role.title standardized and Role.end_datetime adjusted as necessary
+            such that "Councilmember" roles have non-overlapping start_/end_datetime
+        """
+        if not roles:
+            return roles
+
+        if not chair_aliases:
+            chair_aliases = "(chair)|(supervisor)"
+        else:
+            # (chair)|(supervisor)|(foo)|(bar)
+            chair_aliases = f"(chair)|(supervisor)|({')|('.join(chair_aliases)})"
+
+        if not council_member_aliases:
+            council_member_aliases = "council.*member"
+        else:
+            council_president_aliases = (
+                f"(council.*member)|({')|('.join(council_president_aliases)})"
+            )
+
+        if not council_president_aliases:
+            council_president_aliases = "president"
+        else:
+            council_president_aliases = (
+                f"(president)|({')|('.join(council_president_aliases)})"
+            )
+
+        class StdTitles(NamedTuple):
+            title: str
+            pattern: str
+
+        # std role names and patterns to use to match
+        # NOTE: TODO: will use cdp_backend.databse.constants.RoleTitle
+        # when we get a new cdp_backend release
+        std_titles: List[StdTitles] = [
+            # search in this order, e.g. look for vice chair before chair
+            # i.e. more specific search tokens first
+            StdTitles("Vice Chair", "vice.*chair"),
+            StdTitles("Chair", chair_aliases),
+            StdTitles("Councilmember", council_member_aliases),
+            StdTitles("Member", "member"),
+            StdTitles("Council President", council_president_aliases),
+            StdTitles("Alternate", "alternate"),
+        ]
```

Comment:
We can review and approve this PR but I wouldn't merge until cdp-backend new version is released. Which will be later today :slightly_smiling_face: